integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/UnknownEntryFacetListener.java

```diff
@@ -0,0 +1,60 @@
+package marquez.spark.agent.lifecycle.plan;
+
+import static java.util.Optional.ofNullable;
+import static scala.collection.JavaConversions.mapAsJavaMap;
+import static scala.collection.JavaConversions.seqAsJavaList;
+
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import marquez.spark.agent.facets.UnknownEntryFacet;
+import marquez.spark.agent.facets.UnknownEntryFacet.FacetEntry;
+import org.apache.spark.sql.catalyst.expressions.AttributeReference;
+import org.apache.spark.sql.catalyst.expressions.AttributeSet;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.types.DataType;
+import scala.collection.JavaConversions;
+
+public class UnknownEntryFacetListener implements Consumer<LogicalPlan> {
+
+  private final Map<LogicalPlan, Object> processedNodes = new IdentityHashMap<>();
+
+  @Override
+  public void accept(LogicalPlan logicalPlan) {
+    processedNodes.put(logicalPlan, null);
+  }
+
+  public UnknownEntryFacet build(LogicalPlan root) {
+    FacetEntry output = mapEntry(root);
+    List<FacetEntry> inputs =
+        seqAsJavaList(root.collectLeaves()).stream()
+            .map(this::mapEntry)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    return output == null && inputs.isEmpty() ? null : new UnknownEntryFacet(output, inputs);
+  }
+
+  private FacetEntry mapEntry(LogicalPlan x) {
+    if (processedNodes.containsKey(x)) return null;
+    List<UnknownEntryFacet.AttributeField> output = attributeFields(x.outputSet());
+    List<UnknownEntryFacet.AttributeField> input = attributeFields(x.inputSet());
+    return new FacetEntry(x.toJSON(), input, output);
```

Comment:
The logical plan visitor already gets the output of this `toJSON` call. I think what we'll need is something beyond the built-in `toJSON` converter- maybe something that includes that information, but also adds the contents of private fields? Especially if those private fields are themselves LogicalPlan implementations (e.g., a subquery contains a reference to the LogicalPlan of the query).

Reply:
it's good idea, will take a look on that. We want to have similar JSON format. right?

Reply:
@collado-mike  maybe you have some example, how can I test that?